packages/xsnap/test/xsnap-eof.test.js

```diff
@@ -0,0 +1,290 @@
+        { stdio: ['ignore', 'pipe', 'pipe', 'ipc'] },
+      )
+    );
+
+  const outP = text(child.stdout);
+  const errP = text(child.stderr);
+
+  child.on('error', err => {
+    exitedPKit.reject(err);
+  });
+
+  child.on('exit', (code, signal) => {
+    exitedPKit.resolve({ code, signal });
+  });
+
+  const [strOut, strErr, { code: nodeExitCode, signal: nodeExitSignal }] =
+    await Promise.all([outP, errP, exitedPKit.promise]);
+
+  t.is(strOut, '', 'stdout must be empty');
+  t.regex(
+    strErr,
+    /Has parent died\?/,
+    'stderr must contain "Has parent died?"',
+  );
+  t.is(nodeExitCode, null, 'exit code must be null');
+  t.is(nodeExitSignal, 'SIGKILL', 'exit signal must be "SIGKILL"');
+});
+
+async function spawnReflectiveWorker(handleCommand) {
+  const exitedPKit = makePromiseKit();
+  /** @type {XsnapChildProcess | undefined} */
+  let xsnapProcess;
+
+  /** @type {typeof import('child_process').spawn} */
+  const spawnSpy = (...args) => {
+    // @ts-expect-error overloaded signature
+    const cp = proc.spawn(...args);
+    xsnapProcess = cp;
+    return cp;
+  };
+
+  const io = { spawn: spawnSpy, os: os.type(), fs, tmpName };
+  const vat = await xsnap({
+    ...options(io),
+    handleCommand,
+    stdout: 'pipe',
+    stderr: 'pipe',
+  });
+  await vat.evaluate(
+    'function handleCommand(message) { issueCommand(new Uint8Array().buffer); };',
+  );
+
+  assert(xsnapProcess);
+
+  const toXsnap = xsnapProcess.stdio[3];
+  const fromXsnap = xsnapProcess.stdio[4];
+  const stderrP = text(xsnapProcess.stderr);
+  const stdoutP = text(xsnapProcess.stdout);
+
+  xsnapProcess.on('error', err => {
+    exitedPKit.reject(err);
+  });
+
+  xsnapProcess.on('exit', (code, signal) => {
+    exitedPKit.resolve({ code, signal });
+  });
+
+  return {
+    vat,
+    xsnapProcess,
+    toXsnap,
+    fromXsnap,
+    stdoutP,
+    stderrP,
+    exitedP: exitedPKit.promise,
+  };
+}
+
+async function issueCommandAndWait(worker, beforeWait, afterWait) {
+  let beforeWaitError;
+  let afterWaitError;
+  const { stdoutP, stderrP, exitedP } = worker;
+
+  try {
+    // this may error out since we are intentionally
+    // breaking communications with the worker
+    await beforeWait(worker); // vat.issueStringCommand('0');
+  } catch (err) {
+    beforeWaitError = err;
+  }
+
+  // wait for the worker to exit
+  const [strOut, strErr, { code: vatExitCode, signal: vatExitSignal }] =
+    await Promise.all([stdoutP, stderrP, exitedP]);
+
+  try {
+    // this may error out since vat has died.
+    await afterWait(worker); // vat.close();
+  } catch (err) {
+    afterWaitError = err;
+  }
+  return {
+    beforeWaitError,
+    afterWaitError,
+    strOut,
+    strErr,
+    vatExitCode,
+    vatExitSignal,
+  };
+}
+
+async function issueCommandZero(worker) {
+  await worker.vat.issueCommand('0');
+}
+
+async function closeVat(worker) {
+  await worker.vat.close();
+}
+
+async function verifyStdError(t, results, expectedStderr) {
```

Comment:
All verification in this file can be synchronous.
```suggestion
function verifyStdError(t, results, expectedStderr) {
```